### DISCUSSION TITLE
downloading '.zip' file from API but file is damaged/corrupted

### DISCUSSION THREAD

**Discussion by L-S-ONG** [quasarframework/quasar#10231] [Quasar App Extensions] [answered]
Hi fellow quasar peeps, I am currently trying to download .zip file generated from my API, but I'm getting error message saying "The archive is either in unknown format or damaged" whenever I open the file.

```
<script>
import { exportFile } from 'quasar'

export default {
  methods: {
    downloadCSV () {
      this.$axios.get('/download',
        {
          params: {
            filename: this.$route.query.filename
          }
        },
        {
          responseType: 'blob'
        })
        .then(function (response) {
          const status = exportFile(this.$route.query.filename, response.data)
          if (status) {
            // browser allowed it
            console.log('ok')
          } else {
            // browser denied it
            console.log('Error: ' + status)
          }
        }.bind(this))
    }
  },
  mounted () {
    this.downloadCSV()
  }
}
</script>
```

I've tried various solutions such as using
```responseType: 'arraybuffer'```

or

```
headers: {
  'Content-Type': 'multipart/form-data'
},
```

or 

```
const status = exportFile(this.$route.query.filename, response.data, {
  mimeType: 'application/zip'
})
```

reference to: https://quasar.dev/quasar-utils/other-utils#export-file

**Comment by hawkeye64** [comment, marked as answer]
I found this one, which interestingly enough, uses a `responseType` of `"stream"`
https://gist.github.com/senthilmpro/04ea6c086bcb592beb8083214a630548

**Reply by L-S-ONG** [reply]
I've just tried using ```responseType: 'stream'```  it seems like I'd need to use .pipe() on my ```response.data``` and so I tried the link given but it didn't work, still getting damaged/corrupted file

**Reply by MilosPaunovic** [reply]
Check this out:
https://medium.com/js-dojo/force-file-download-in-vuejs-using-axios-a7fe1b5dfe7b

Perhaps you're gonna have to adjust backend endpoint as well.

**Reply by L-S-ONG** [reply]
after several attempts in trying to resolve this issue, I realised that the problem lies on the headers `content-type`

adding this solves the problem
```
          headers: {
            'Content-Type': 'application/json; application/octet-stream'
          }
```

nonetheless, many thanks to @hawkeye64 and @MilosPaunovic
